server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -154,6 +158,36 @@ public Query termsQuery(List<?> values, QueryShardContext context) {
             }
         }
 
+        @Override
+        public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper, QueryShardContext context) {
+            DEPRECATION_LOGGER.deprecatedAndMaybeLog("range_single_type",
+                    "Running [range] query on [_type] field for an index with a single type. As types are deprecated, this functionality will be removed in future releases.");
+            if ((lowerTerm != null && lowerTerm instanceof BytesRef == false)) {
+                throw new IllegalArgumentException("lower term should be BytesRef but was: " + lowerTerm.getClass().getName());
+            }
+            if ((upperTerm != null && upperTerm instanceof BytesRef == false)) {
+                throw new IllegalArgumentException("upper term should be BytesRef but was: " + upperTerm.getClass().getName());
+            }
```

Comment:
Conversion to bytesref is done elsewhere with `indexedValueForSearch`. I'm unsure of the impact of rejecting anything but bytesrefs.